source/distributed/shared_tria.cc

```diff
@@ -57,6 +57,18 @@ namespace parallel
     template <int dim, int spacedim>
     void Triangulation<dim,spacedim>::partition()
     {
+#ifdef DEBUG
+      // Check that all meshes are the same (or at least have the same
+      // total number of active cells):
+      const unsigned int max_active_cells
+        = Utilities::MPI::max(this->n_active_cells(), this->get_communicator());
+      Assert(max_active_cells == this->n_active_cells(),
+             ExcMessage("A shared::Triangulation needs to be refined in the same "
+                        "way on all processors, but there are currently a different "
```

Comment:
The phrasing sounds (grammatically) wrong. Maybe 
`[...], but the participating processors don't agree on the number of active cells.`
or
`[...], but there are currently different numbers of active cells on the participating processors.`

Reply:
I am also fine if you fix this up in a subsequent PR if you prefer that.